File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string myString, string pat) {
    int answer = 0;
    int size = pat.size();

    for (int i = 0; i <= myString.size() - size; i++) {
        if (myString.substr(i, size) == pat) {
            answer++;
        }
    }

    return answer;
}

File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.swift
import Foundation

func solution(_ myString:String, _ pat:String) -> Int {
    
    var result = 0
    var str = myString

    while !str.isEmpty {
        if str.prefix(pat.count).contains(pat) { result += 1 }
        str.removeFirst()
    }
    
    return result
}